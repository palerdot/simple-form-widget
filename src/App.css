.App {
  text-align: center;
}

.App-header {
  background-color: blue;
  height: 50px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  align-self: center;
}

.deal-form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  align-self: center;
  width: 500px; 
}

.steps-header {
  margin: 29px 0;
}

.steps-header span {
  width: 150px;
  height: 50px;
  padding: 7px;
  margin: 7px;
  background-color: white;
  color: black;
  border: thin solid blue;
}

.steps-header span.active {
  background-color: blue;
  color: white;
}

.error-msg {
  color: red;
  font-size: 12px;
  text-align: left;
}
