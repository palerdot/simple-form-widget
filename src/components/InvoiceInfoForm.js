// will be used to show the info of step2 alongside with step1
import React from 'react'

import _ from 'lodash'

import moment from 'moment'

import styles from '../utils/styles'

const InvoiceInfoForm = function (props) {
  return (
    <div style={{display: 'flex', flex: 1, flexDirection: 'column', marginLeft: '29px'}}>
      <h3>INVOICE INFO</h3>

      <div style={styles.dateLabel}>Issue Date:</div>
      <div style={styles.infoValue}>{props.issue_date ? moment(props.issue_date).format('DD-MM-YYYY') : ''}</div>

      <div style={styles.dateLabel}>Repayment Date:</div>
      <div style={styles.infoValue}>{props.repayment_date ? moment(props.repayment_date).format('DD-MM-YYYY') : ''}</div>

      <div style={styles.dateLabel}>Amount:</div>
      <div style={styles.infoValue}>{props.amount}</div>
    </div>
  )
}

export default InvoiceInfoForm