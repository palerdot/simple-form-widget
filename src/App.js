import React, { Component } from 'react'
import './App.css'

// required for material ui
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

// import DealFrom
import Form from './components/Form'

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Deal Creation</h1>
        </header>
        <div className="App-intro">
          <MuiThemeProvider>
            <Form />
          </MuiThemeProvider>
        </div>
      </div>
    );
  }
}

export default App
