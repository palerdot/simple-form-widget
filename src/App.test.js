import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import renderer from 'react-test-renderer'

// import our step creation components
import InvoiceInfoForm from './components/InvoiceInfoForm'

// import validators
import { 
  amountValidator, 
  nameValidator, 
  invoiceAmountValidator,
  dateValidator,
  repaymentDateValidator,
} from './utils/validators'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('should render info form', () => {
  const component = renderer.create(
    <InvoiceInfoForm 
      name='step 2 name'
      amount={23}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// sample tested
// test if it is a valid name
test('should test if given name is valid', () => {
  let valid_name = 'valid step two name'
  expect(nameValidator(valid_name)).toBe(true)
})

test('should test if given name is invalid', () => {
  let invalid_name = 'valid step 2 name'
  expect(nameValidator(invalid_name)).toBe(false)
})

test('invoice amount should be greater than deal amount', () => {
  let invoice_amount = 25
  let deal_amount = 21
  expect(invoiceAmountValidator(invoice_amount, deal_amount)).toBe(true)
})
