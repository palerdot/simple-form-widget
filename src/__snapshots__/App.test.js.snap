// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render info form 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "marginLeft": "29px",
    }
  }
>
  <h3>
    INVOICE INFO
  </h3>
  <div
    style={
      Object {
        "color": "#ABABAB",
        "fontSize": 15,
        "marginTop": "29px",
        "textAlign": "left",
      }
    }
  >
    Issue Date:
  </div>
  <div
    style={
      Object {
        "color": "#ABABAB",
        "fontSize": 17,
        "textAlign": "left",
      }
    }
  >
    
  </div>
  <div
    style={
      Object {
        "color": "#ABABAB",
        "fontSize": 15,
        "marginTop": "29px",
        "textAlign": "left",
      }
    }
  >
    Repayment Date:
  </div>
  <div
    style={
      Object {
        "color": "#ABABAB",
        "fontSize": 17,
        "textAlign": "left",
      }
    }
  >
    
  </div>
  <div
    style={
      Object {
        "color": "#ABABAB",
        "fontSize": 15,
        "marginTop": "29px",
        "textAlign": "left",
      }
    }
  >
    Amount:
  </div>
  <div
    style={
      Object {
        "color": "#ABABAB",
        "fontSize": 17,
        "textAlign": "left",
      }
    }
  >
    23
  </div>
</div>
`;
